pkg/virt-controller/watch/snapshot/restore.go

```diff
@@ -598,18 +630,55 @@ func (ctrl *VMRestoreController) getSnapshotContent(vmRestore *snapshotv1.Virtua
 	return vmss, nil
 }
 
-func (ctrl *VMRestoreController) getVM(namespace, name string) (*kubevirtv1.VirtualMachine, error) {
+func (ctrl *VMRestoreController) getVM(namespace, name string) (vm *kubevirtv1.VirtualMachine, exists bool, err error) {
 	objKey := cacheKeyFunc(namespace, name)
 	obj, exists, err := ctrl.VMInformer.GetStore().GetByKey(objKey)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 
 	if !exists {
-		return nil, fmt.Errorf("VirtualMachine %s/%s does not exist", namespace, name)
+		return nil, false, nil
 	}
 
-	return obj.(*kubevirtv1.VirtualMachine).DeepCopy(), nil
+	return obj.(*kubevirtv1.VirtualMachine).DeepCopy(), true, nil
+}
+
+func patchVM(vm *kubevirtv1.VirtualMachine, patches []string) (*kubevirtv1.VirtualMachine, error) {
+	if len(patches) == 0 {
+		return vm, nil
+	}
+
+	log.Log.V(3).Object(vm).Infof("patching restore target. VM: %s. patches: %+v", vm.Name, patches)
+
+	if vm.ObjectMeta.Name == "" || vm.ObjectMeta.Namespace == "" {
+		vm.ObjectMeta = vm.Spec.Template.ObjectMeta
```

Comment:
Don't think we want to do this and honestly I'm surprised it's not causing an issue when restoring to new VM as namespace will be blank

Reply:
Why do you think namespace would be blank? The opposite happened here, `vm.ObjectMeta` had missing values.

And in general - doesn't `vm.Spec.Template.ObjectMeta` supposed to reflect `vm.ObjectMeta`?

Would you prefer that I'd only copy `Name` and `Namespace` here?